File: liquid/clientTemplateValue/dotnet/ClientTemplateValue.cs
using DotLiquid;

namespace LitQuid.Liquid.Filters
{
    /// <summary>
    /// LitQuid Liquid filter for marking client-side template values.
    /// Usage: {{ value | clientTemplateValue: "${this.propertyName}" }}
    /// </summary>
    public static class ClientTemplateValue
    {
        /// <summary>
        /// A noop filter that acts as a marker for build-time extraction.
        /// This allows Liquid to parse the template safely while marking
        /// which values should be extracted for client-side Lit templates.
        /// </summary>
        /// <param name="input">The input value to pass through.</param>
        /// <param name="clientExpression">The client-side expression (ignored at runtime, used by preprocessor).</param>
        /// <returns>The string representation of the input, or empty string if null.</returns>
        public static string ClientTemplateValueFilter(object input, string clientExpression = "")
        {
            // The clientExpression is only used by the Rust preprocessor
            // At runtime, we just return the server-rendered value
            return input?.ToString() ?? string.Empty;
        }

        /// <summary>
        /// Registers the filter with DotLiquid.
        /// Call this before rendering templates.
        /// </summary>
        public static void Register()
        {
            Template.RegisterFilter(typeof(ClientTemplateValue));
        }
    }
}

File: liquid/clientTemplateValue/java/src/main/java/com/litquid/liquid/filters/ClientTemplateValue.java
package com.litquid.liquid.filters;

import liqp.filters.Filter;

/**
 * LitQuid Liquid filter for marking client-side template values.
 *
 * A noop filter that acts as a marker for build-time extraction.
 * This allows Liquid to parse the template safely while marking
 * which values should be extracted for client-side Lit templates.
 *
 * Usage in .liquid templates:
 * <pre>
 * {{ firstName | clientTemplateValue: "${this.firstName}" }}
 * </pre>
 */
public class ClientTemplateValue extends Filter {

    public ClientTemplateValue() {
        super("clientTemplateValue");
    }

    @Override
    public Object apply(Object value, Object... params) {
        // Pass through the value unchanged
        // params[0] would be the client expression string, but we ignore it at runtime
        return value == null ? "" : value.toString();
    }

    /**
     * Register this filter with the Liqp template engine.
     *
     * @return a new instance of the filter for registration
     */
    public static ClientTemplateValue create() {
        return new ClientTemplateValue();
    }
}
